### DISCUSSION TITLE
Satellite DC Actuator - Reed Switch

### DISCUSSION THREAD

**Discussion by BMOC1** [micropython/micropython#11525] [Hardware & Peripherals]
I have a 12 volt DC actuator.  I know it has a magnetic switch in the motor that sends pulses that are counted to keep track of how long to run the motor.  I would like to create some hardware/software that can detect and count those pulses?

What would i need on the Raspberrry side hardware wise?  My thinking was i could just buy a reed switch : 

Reed Switch Normally Open, Reed Sensor Module Magnetic Switch Module DIY Kit Normally Open for Arduino

![image](https://github.com/micropython/micropython/assets/45899790/8735ae6a-0264-4abe-b220-bedc05db6e27)

and then take out the magnetic switch that is hooked into positive and negative, and then just pipe in the positive and negative that comes from the actuator reed switch.   Would that work?  Is my thinking right? or, is it simpler than that in terms of what i would need just to catch those pulses??

Thanks!

**Comment by peterhinch** [comment]
For anyone interested I found [this link](https://www.ebay.co.uk/itm/304929512466?chn=ps&norover=1&mkevt=1&mkrid=710-153316-527457-8&mkcid=2&itemid=304929512466&targetid=4585513251572427&device=c&mktype=&googleloc=&poi=&campaignid=431353846&mkgroupid=1310618283030814&rlsatarget=pla-4585513251572427&abcId=9301942&merchantid=87779&msclkid=9906273a6f401e113394435efd6f943c) which shows what this module does. The circuit aims to debounce the switch. Alas it was evidently designed by the apprentice who had never heard the term "Schmitt Trigger".

To answer the question, detecting a switch contact closure is easy, but we need to know the rate at which the reed switch will be activated. If there is one pulse per rev of the motor, what is its maximum RPM?

I'm also unclear about what you have in mind. Are you saying that the motor outputs a 12V pulse, or that the shaft has a magnet that activates the reed switch? The board you have will not tolerate 12V.

**Comment by 2dof** [comment]
dc motor in  Satellite DC Actuator  will usually operate with speed 3000-6000 rpm  max , so worst case scenario it give 100 Hz pulses (one magnet on shaft).   Your Module is based on reed switch and there will be problems with that high switching frequency(reed switch are usually used for low frequency or just open/closes sensing), .

 I think it is better use a HAL sensor solution instead 

 edit: @peterhinch  i think  @OscamSatUser is using a 12 dc motor (common solution for satellite dc actuators) , but i guess a construction without build in sensor is used ( build in solution use HALs) . These drives are quite compact and installation sensor and magnet can be problematic ( sometimes construction has already space for sensor unit but not used in product but depends form manufacturer)

**Comment by BMOC1** [comment]
I've been trying to get the specs on the actuator in terms of rotational speed.... Yes, its a 12 volt DC motor that moves the actuator.. But, that's not what runs through the sensor.  I did however have an extra sensor..

Super Sensor II - VMC 8101 HTS

I read this : We can provide you these types of satellite actuators in various options including Potentiometer feedback 0.4 kpi, Super Sensor II Reed Switch 32 ppi, and Super Sensor II Hall Effect 32 ppi based on your requirements.  I'm not sure which of these two refers to the VMC unit. hope this helps.

**Reply by 2dof** [reply]
@OscamSatUser  You said that "it has a magnetic switch in the motor" but also  "that's not what runs through the sensor " so that  means sensor is placed on the actuator side (piston side)?    

In industrial linear actuators reed switches are used as proximity (max/min position)sensors;  for speed: Hall/potentiometers/Optic sensing is used.  I meet a reed switch sensing but only in piston side ( low RPM side), never on motor (shaft) side.

also "and then take out the magnetic switch that is hooked into positive and negative, and then just pipe in the positive and negative that comes (..)"  -> positive and negative of what? 

if You can clarify  it will be helpful.

**Comment by BMOC1** [comment]
8vdc is a typical sensor voltage, but told they can be anywhere from 5-12vdc
it's a reed switch that provide an open / close cycle for a circuit.

The switch is connected next to the actuator motor that is rotating.  Then there is shielded 1 pair wire that runs to the device utilizes the pair wire to detect the pulses.

Is this enough information, if not, what additional info do you need?

**Reply by 2dof** [reply]
" actuator motor that is rotating" - do You mean  motor shaft  (High rpm side)  or  piston side (screw shaft - low rpm shaft)?  from examples photos form sellers  usually on piston side (after speed reduction). It You have like that then  KIT from Your photo will be sufficient if not (on motor shaft side) then I do not recommend that solution.

**Reply by BMOC1** [reply]
I would assume the low rpm shaft, but, ill have to open up one of the actuators to check.. Ill have to wait for some clear weather.. Ill post back.

**Comment by BMOC1** [comment]
It took me a bit of time, as the weather outside wasn't nice.  I took a picture... The sensor is on the actuator side, not the motor side. The bottom view, which is what i took a picture of has the actual ROD arm on the opposite side of the sensor and magnet.


![image](https://github.com/micropython/micropython/assets/45899790/108be91c-6e6c-4a0b-81c1-9044d41dd6d4)

**Reply by 2dof** [reply]
Magnetic Switch Module DIY Kit from Your fot will be sufficient. Im not how many magnets you have on rotating disc but since it is low speed then even with fro. 1 to 4  shoulb be ok. 
On tjr photo you have reedswitch in plastic package, if it woking then jus connect it to pcb of kit ( replace with kit switch) . If oryginal is damaged then just replace,).
I guess thalt counting you will be using esp or other microcontroller with mixropython, then use @peterhinch  asyncio button driver lib ( great for debouncing).

**Comment by zcattacz** [comment]
If you want to use the existing couter's signal output (5~12V?), use some resister + an [optical isolator](https://www.sparkfun.com/products/9118) will be enough to protect the RPi's IO.

**Comment by BMOC1** [comment]
How would i find out what the voltage is on it?  Okay, i might be asking a dumb question, but i thought a reed switch just opens/closes the circuit.  Many made mention of voltage, but i wouldn't think there is any voltage on the line other than what your sending through on the detect side to see if there is continuity or not, in the line...  

At the moment, I'm not worried about powering the motor.. Right now, all i want to do is reliably count the cycles coming through on the reed switch...

**Reply by 2dof** [reply]
Reed Sensor Module Magnetic Switch Module DIY Kit  (from Your photo) has power from 3.3 to 5 V ( just like microcontroller ). pcb has 3 pins : Vcc , Gnd  and DO (digital output directly connected to DI of microcontroller) .  

At the begging ( when You will be developing software) use power supply from microcontroller board (connected to usb/charger). 
also resolder your reed swtich:  pcb-> 2 wires -> to reed switch (best if You can use that from photo (at least reuse plastic housing)) . 
When Your software will be done,  then adapt power supply from 12 V ( if from car battery) or just use source best fit for Your setup.
